### PR TITLE
Test sources as file

### PR DESCRIPTION
**Pull request by RanjanarMS**

### CODE REVIEW

File: Tasks/VsTest/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -106,6 +106,7 @@
   "loc.messages.testplanConfigInput": "Test plan configuration Id : %s",
   "loc.messages.testSuiteSelected": "Test suite Id selected: %s",
   "loc.messages.testAssemblyFilterInput": "Test assemblies : %s",
+  "loc.messages.testAssemblyFilterForTestPlan": "Setting the test assembly filter to '%s' for the 'Test plan'",
```
Comment (nigurr) [inline Tasks/VsTest/Strings/resources.resjson/en-US/resources.resjson]: remove

File: Tasks/VsTest/distributedtest.ts [hunk 2]
```diff
@@ -140,23 +145,52 @@ export class DistributedTest {
...
+            if (this.testSourcesFile) {
+                tl.rmRF(this.testSourcesFile);
+            }
+        } catch (error) {
+            //Ignore.
+        }
+        this.dtaPid = -1;
+    }
+
+    private getTestSourcesFile() : string {
```
Comment (nigurr) [inline Tasks/VsTest/distributedtest.ts]: createTestSources

File: Tasks/VsTest/distributedtest.ts [hunk 3]
```diff
@@ -140,23 +145,52 @@ export class DistributedTest {
...
+        try {
+            const sources = tl.findMatch(this.dtaTestConfig.testDropLocation, this.dtaTestConfig.sourceFilter);
+            tl.debug('tl match count :' + sources.length);
+            const filesMatching = [];
+            sources.forEach(function(match: string) {
+                if (!fs.lstatSync(match).isDirectory()) {
```
Comment (nigurr) [inline Tasks/VsTest/distributedtest.ts]: ERR handling

File: Tasks/VsTest/distributedtest.ts [hunk 4]
```diff
@@ -140,23 +145,52 @@ export class DistributedTest {
...
+                    filesMatching.push(match);
+                }
+            });
+
+            tl.debug('Files matching count :' + filesMatching.length);
+            if (utils.Helper.isNullOrUndefined(filesMatching) || filesMatching.length === 0) {
```
Comment (nigurr) [inline Tasks/VsTest/distributedtest.ts]: remove

File: Tasks/VsTest/distributedtest.ts [hunk 5]
```diff
@@ -140,23 +145,52 @@ export class DistributedTest {
...
+                throw new Error(tl.loc('noTestSourcesFound', this.dtaTestConfig.sourceFilter.toString()));
+            }
+
+            const tempFile = path.join(os.tmpdir(), 'testSources_' + uuid.v1() + '.src');
+            fs.writeFileSync(tempFile, filesMatching.join('\n'));
```
Comment (nigurr) [inline Tasks/VsTest/distributedtest.ts]: os.EOL

File: Tasks/VsTest/taskinputparser.ts [hunk 6]
```diff
@@ -175,6 +178,35 @@ function initTestConfigurations(testConfiguration: models.TestConfigurations) {
...
+
+    const isEnabledInHKCU = await isDontShowUIRegKeySet(regPathHKCU);
+    const isEnabledInHKLM = await isDontShowUIRegKeySet(regPathHKLM);
+
+    if (!isEnabledInHKCU && !isEnabledInHKLM) {
+        tl.warning(tl.loc('DontShowWERUIDisabledWarning'));
+    }
+}
+
+function isDontShowUIRegKeySet(regPath: string): Q.Promise<boolean>  {
```
Comment (nigurr) [inline Tasks/VsTest/taskinputparser.ts]: avoid Q

File: Tasks/VsTest/Strings/resources.resjson/en-US/resources.resjson (hunk 1 above)
Comment (RanjanarMS) [inline Tasks/VsTest/Strings/resources.resjson/en-US/resources.resjson]: Done

File: Tasks/VsTest/distributedtest.ts (hunk 2 above)
Comment (RanjanarMS) [inline Tasks/VsTest/distributedtest.ts]: Done

File: Tasks/VsTest/distributedtest.ts (hunk 3 above)
Comment (RanjanarMS) [inline Tasks/VsTest/distributedtest.ts]: Not available lstatsync.. And we throwing exception back..

File: Tasks/VsTest/distributedtest.ts (hunk 4 above)
Comment (RanjanarMS) [inline Tasks/VsTest/distributedtest.ts]: Done

File: Tasks/VsTest/distributedtest.ts (hunk 5 above)
Comment (RanjanarMS) [inline Tasks/VsTest/distributedtest.ts]: Done

File: Tasks/VsTest/taskinputparser.ts (hunk 6 above)
Comment (RanjanarMS) [inline Tasks/VsTest/taskinputparser.ts]: Not convinced :-)

**Pull request merged**